MozziGuts.cpp

```diff
@@ -346,20 +417,53 @@ void audioHook() // 2us excluding updateAudio()
 	}
 //setPin13Low();
 }
-
+        
+#if IS_SAMD21()
+        void TC5_Handler (void) __attribute__ ((weak, alias("samd21AudioOutput")));
+#endif
 
 
 //-----------------------------------------------------------------------------------------------------------------
 #if (AUDIO_MODE == STANDARD) || (AUDIO_MODE == STANDARD_PLUS) || IS_STM32()
-
-#if IS_TEENSY3()
+#if IS_SAMD21()
+#ifdef __cplusplus
+    extern "C" {
+#endif
+        
+        void samd21AudioOutput(void);
+#ifdef __cplusplus
+    }
+#endif
+    
+#elif    IS_TEENSY3()
   IntervalTimer timer1;
 #elif IS_STM32()
   HardwareTimer audio_update_timer(AUDIO_UPDATE_TIMER);
   HardwareTimer audio_pwm_timer(AUDIO_PWM_TIMER);
+
 #endif
 
-#if IS_TEENSY3()
+#if IS_SAMD21()
+#ifdef __cplusplus
+    extern "C" {
+#endif
+        
+     void samd21AudioOutput()
+    {
+        
+#if (USE_AUDIO_INPUT==true)
+        adc_count = 0;
+        startSecondAudioADC();
+#endif
+        
+        analogWrite(DAC0, (int)output_buffer.read());
```

Comment:
I suggest using a define in place of DAC0 (even if it may not be user configurable). That will be more in line with the other architectures. Also, this function looks pretty similar to the Teensy3 one. Could they merged?

Reply:
They could be merged but this way there is room for I2S support which is likely different between SAMD21 and Teensy.

Reply:
It may be that DAC0 is now the "Official" Arduino way of naming the DAC pin. I am loath to create yet another name. None of these names work very well anyway because it is called A0/D0/~ for example on Arduino Gemma. The ~ is the most helpful actually.

Reply:
What I means is

```
#define AUDIO_CHANNEL_1_PIN DAC0
```

somewhere. That is used consistently _inside Mozzi_.

Reply:
ok

Reply:
Since you have a platform specific place for such defines, why not put the DAC range in there too? e.g. DACRANGE 4096, 2048, 256. Then the output calls can have something like * DACRANGE/256  (or DACRANGE/ 65536 with a 16-bit table). These would be optimized into shifts by most compilers...

Reply:
Yes, you are right, something along those lines will be needed, consistently for all platforms. You may want to comment on my proposal at https://github.com/sensorium/Mozzi/issues/43 . Not sure this is the best way to do it, hence the ticket for discussing it, first.